components/crossplane-control-plane/staging/provider-config.yaml

```diff
@@ -0,0 +1,38 @@
+---
+apiVersion: kubernetes.crossplane.io/v1alpha1
+kind: ProviderConfig
+metadata:
+  name: eaas-stage-kubernetes-provider-config
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: eaas-cluster
+      key: kubeconfig
+---
+apiVersion: external-secrets.io/v1alpha1
+kind: ExternalSecret
+metadata:
+  name: eaas-cluster
+  namespace: crossplane-system
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    argocd.argoproj.io/sync-wave: "-1"
+spec:
+  dataFrom:
+    - extract:
+        key: staging/eaas/eaas-stage-kubeconfig
+  refreshInterval: 1h
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: appsre-stonesoup-vault
+  target:
+    creationPolicy: Owner
+    deletionPolicy: Delete
+    name: eaas-cluster
+    template:
+      data:
+        apiserver: https://api.kflux-stg-es01.21tc.p1.openshiftapps.com:6443
```

Comment:
This should be stored in vault in the same secret as the kubeconfig.

Reply:
Updated